### PR TITLE
Added help tab to main menu

### PR DESCRIPTION
**Pull request by TheShadowZone12**
Added a help tab to main menu to assist new players and provide informational links.

### CODE REVIEW

File: builtin/mainmenu/tab_help.lua [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				"," ..
+				"http://wiki.minetest.net/Getting_Started," ..
+				"," ..
+				"#FFFF00" .. fgettext("Minetest Wiki") .. "," ..
+				"http://wiki.minetest.net," ..
+				"," ..
+				"#FFFF00" .. fgettext("Minetest IRC help and chat") .. "," ..
+				"Freenode #minetest," ..
+				"Development channel #minetest-dev," ..
+				"Inchra.net #minetest," ..
```
Comment (ShadowNinja) [inline builtin/mainmenu/tab_help.lua]: This isn't currently an official Minetest channel.

That might not matter though.

File: builtin/mainmenu/tab_help.lua [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				"," ..
+				"#FFFF00" .. fgettext("Minetest Forums") .. "," ..
+				"http://forum.minetest.net," ..
+				"," ..
+				"#FFFF00" .. fgettext("Minetest downloads") .. "," ..
+				"For Downloads for most OS, Android or a newer version," ..
```
Comment (ShadowNinja) [inline builtin/mainmenu/tab_help.lua]: The comma here causes a new line.  Also, if you have Minetest you probably already know this.

**Comment by ShadowNinja** [comment]
The headers use fgettext, but the rest of the text doesn't, so it isn't translatable.

I also think the logo, version, etc from the credits is unnecessary.

Screenshot please?  :-)

**Comment by crazyginger72** [comment]
I do like the idea and the design of the logo and player :) and version might be useful as remember some of the users of minetest are not computer literate!


help_tab

**Comment by kilbith** [comment]
Also, you don't deserve to be in credits. I find quite questionable to honor yourself and especially for one pull and a basic descriptive code...

**Comment by TheShadowZone12** [comment]
There ya go.

thank you crazyginger72.

**Comment by crazyginger72** [comment]
@kilbith it is under contrubuitors and this is a contribution i dont see the issue!

**Comment by kilbith** [comment]
@crazyginger72  So you would see no objection to add a hundred of contributors in credits then ? ;)

**Comment by crazyginger72** [comment]
@kilbith i have deved projects before with a contributor like 800 lines long, ANYONE WHO CONTRIBUTES MUST BE RECOGNIZED!!!!!

**Comment by TheShadowZone12** [comment]
If you want me to change any of the files I will willingly change them, I have no objections to removing myself from the credits if it is gonna cause conflict.

**Comment by ShadowNinja** [comment]
The rule for "Active contributor" is "more than one commit within the last three months", but there are some other people missing there, like paramat.

**Comment by TheShadowZone12** [comment]
There, removed....now let's get to the real purpose of this pull request the help tab.

**Comment by ShadowNinja** [comment]
After seeing the screenshot I think that
  1. The images and text on the left should be removed.  The version is in the top left corner and that's the only really helpful item other than possibly the URL, which can be placed elsewhere.
  2. The lines should be made longer (especially after removing the stuff on the left).  If line wrapping works well you should put each paragraph on one line (although it should be wrapped to ~80 chars in the code).

**Comment by kilbith** [comment]
IMO, the right place of that superfluous pull is in trash for obvious reasons...
- /help command is available.
- The controls configuration is already displayed in settings.
- All the basic documentation / contact is easily findable on minetest.net.
- The design is awful.

**Comment by ShadowNinja** [comment]
@kilbith I disagree.  The text may need improvement, but the concept of having a help menu is good, I looked for one myself when I first played Minetest.

**Comment by crazyginger72** [comment]
@kilbith  F YOU!! YOUR ATTACKING A CHILD FOR TRYING TO HELP OUT, YOU SHOULD BE PROUD SOME ONE WANTS TO HELP OUT AT ALL!!!!

**Comment by TheShadowZone12** [comment]
@ShadowNinja that is exactly why I made it in the first play. cause I also looked for one the first time I joined.

**Comment by MT-Modder** [comment]
@kilbith You have to remember this is for people who know very little. For people where it's a miracle that they found the executable(windows)  :smiley:

File: builtin/mainmenu/tab_help.lua [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+				" for a specific command. ," ..
+				"," ..
+				"http://wiki.minetest.net/Getting_Started," ..
+				"," ..
+				"#FFFF00" .. fgettext("Minetest Wiki") .. "," ..
+				"http://wiki.minetest.net," ..
+				"," ..
+				"#FFFF00" .. fgettext("Minetest IRC help and chat") .. "," ..
+				"Freenode #minetest," ..
+				"Development channel #minetest-dev," ..
```
Comment (ShadowNinja) [inline builtin/mainmenu/tab_help.lua]: This doesn't specify a network.  Also, it should be clear that -dev is strictly for core development, not mods or the like.

Other way to do this which I would prefer:
```
Main channel: #minetest on freenode.net
Core development channel: #minetest-dev on freenode.net
Server chat channel: #minetest on inchra.net
```

**Comment by deezl** [comment]
@kilbith, wow, what a miserable thing to say to someone who is trying to come up with ways to make the game better. I think a simple help tab with basic instructions and a nice organized place to point new users to helpful documentation is quite welcome. The design is simply based on the overall design of the client (which isn't all that attractive to begin with.) It's not adding any new design concepts, she didn't invent the design, just used it. If you are going to criticize, be helpful, not simply blast it. I for one encouraged shawzone to proceed with this idea when she first mentioned it. How about, if you think it needs more, or less even, concetrate on helping her to make it better.

**Comment by TheShadowZone12** [comment]
@deezl Thank you! I really appreciate the helpful encouraging words.

**Comment by kilbith** [comment]
@deezl : which is miserable is to think that I'm attacking the intention of helping players or shadowzone herself. This is not correct. I'm just against the redundant concept suggested here and I simply explained why above.

About the design, I was referring to the placing / formatting of the elements, i.e. the images and the text (tight column), and this is not related to the overall formspec design itself.

If some people absolutely want this added, I assume that'd need a perseverant work for make it pleasant and useful.

**Comment by crazyR14** [comment]
@TheShadowZone12 i love it, its needed and i hope it gets added. ive lost track of how many players come to my server with out even basic knowledge of how to do things described in your help tab.

@kilbith  not sure what is wrong with you today but you definitely came across as attacking @TheShadowZone12 and attacking the very idea behind this pull request, You can deny it all you want but the fact that several people have noticed it and commented regarding it, go to show that you are.

Please be more constructive and friendly.

**Comment by kilbith** [comment]
@crazyR14 : attacking a concept, yes. Is that allowed or not ? :)
However someone told me to "fuck you" just now and nobody has reacted on this, paradoxically...

**Comment by crazyR14** [comment]
@kilbith i refuse to argue with you, if you wasn't so bitter then maybe your would not get people telling you to "fuck you".
I'm not saying that the person who said it is right for saying it but if you talk towards people int he the way you have done above you can expect a negative reaction.

**Comment by kilbith** [comment]
Interesting...

So you create a wrong amalgam as : give an argumented objection on a pull == attacking shadowzone, then you refuse to discuss with me and give a moral caution to the one who insults me...

Who's really not friendly, open-minded and constructive finally ? ;)

**Comment by deezl** [comment]
Oh FFS, this is supposed to be a discussion about adding a help menu to the game. Let the other crap go.

**Comment by crazyR14** [comment]
@kilbith i refuse to arhue because this is not the place to be argueing. and also because i would be wasting my time with you. if you wanter further this, then take it to irc.

**Comment by TheShadowZone12** [comment]
I would appreciate if you'd talk more about the pull request than go after kilbith, crazyR and deezl.

**Comment by crazyginger72** [comment]
@kilbith  i said " F YOU"   i never said what f word, so let me clarify that

FIZZLE YOU!!!!  :)  i didnt swear and if you took it as that i am ferry sorry dear.

@TheShadowZone12  ShadowNinja did suggest the use of the gettext, i believe that is a good idea as it translates if im not mistaken ( probably am)

**Comment by kwolekr** [comment]
The contents here look like they belong in a help.txt rather than being hardcoded in the scripts.  What if you were to make the help tab just read and display the contents of help.txt with some formatting or something?

**Comment by deezl** [comment]
@kwolekr that could be a nice way to present it. 
@TheShadowZone12 what do you think of that idea?

**Comment by deezl** [comment]
Could there be a way for there to be internal links to a document, with an index?

**Comment by TheShadowZone12** [comment]
I like the idea.

**Comment by Zeno-** [comment]
Of course this is a good idea. Once minor "implementation details" are resolved it should be merged.

**Comment by PilzAdam** [comment]
Why is the player image from minetest_game added to the engine? I am against using this.

**Comment by TheShadowZone12** [comment]
So they know what the "defaultskinner" looks like to differentiate them between players with skins.

**Comment by TheShadowZone12** [comment]
The thing is...not many people go and read the code, they download the game and play.  Which is why I thought the "help" tab would've been better.

File: builtin/mainmenu/tab_help.lua (hunk 3 above)
Comment (sfan5) [inline builtin/mainmenu/tab_help.lua]: >Server chat channel: #minetest on inchra.net

This channel is not officially "*approved*" by the dev team, it shouldn't be in there.

**Comment by sfan5** [comment]
>Why is the player image from minetest_game added to the engine? I am against using this.

ack

**Comment by TheShadowZone12** [comment]
@sfan5 I don't understand what you mean..by "ack"

**Comment by sfan5** [comment]
Changed it.

**Comment by TheShadowZone12** [comment]
screenshot from 2015-01-16 10 34 57
New screenshot.

**Comment by ShadowNinja** [comment]
@TheShadowZone12 
  1. You changed `fgettext` to `gettext`, citing me as the reason.  That's not what I said.  I want you to put all of the content in `fgettext` calls so that the whole help menu can be translated, not just the headers.
  2. You simply removed the player icon without moving other things.  This leaves a empty space, which looks wrong.  As I said before though, I think the whole left sidebar should be removed, to make more room for the help and because it isn't very helpfull right now.

**Comment by kilbith** [comment]
Relevant idea from @VanessaE on IRC : http://irc.minetest.ru/minetest-dev/2015-01-16#i_4114097

**Comment by MT-Modder** [comment]
:+1: For @VanessaE's idea.

File: builtin/mainmenu/tab_help.lua [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+--------------------------------------------------------------------------------
+
+tab_help = {
+	name = "help",
+	caption = fgettext("Help"),
+	cbf_formspec = function (tabview, name, tabdata)
```
Comment (sapier) [inline builtin/mainmenu/tab_help.lua]: if you define this function as local function prior defining tab_help you can save a lot of indentions.

**Comment by rubenwardy** [comment]
I would qualify as an active contributor, even though my commits are small XD

**Comment by TheShadowZone12** [comment]
So, does that mean it's getting merged? Or no?

**Comment by rubenwardy** [comment]
I'd personally like a menu to appear on f1. Set the same way as the
inventory - player:set_help_formspec()

**Comment by TheShadowZone12** [comment]
That might be helpful.

**Comment by TheShadowZone12** [comment]
or since "f" doesn't do anything currently, tapping that could bring up a "help" menu.

**Comment by rubenwardy** [comment]
F1 is used in many programs to bring up help. It will configurable and be
on the pause menu.

**Comment by TheShadowZone12** [comment]
VanessaE suggested we add a "help" button to the pause menu.

**Comment by ngosang** [comment]
What about first steps in crafting? http://wiki.minetest.net/How_to_build_your_first_house

**Comment by PilzAdam** [comment]
@ngosang This is a help section for the engine, not the default (or any other) game.

**Comment by TheShadowZone12** [comment]
@PilzAdam I made this to help everyone, I can add whatever helpful links
needed or requested.​

**Comment by TheShadowZone12** [comment]
@rubenwardy I looked, f1 does do something.

**Comment by rubenwardy** [comment]
F1 is universal for help [citation needed]. You could just move what is
there.

**Comment by paramat** [comment]
This would be good to do what is suggested here https://github.com/minetest/minetest/pull/2438 but in a cleaner, optional way.

**Pull request closed without merging**